samplers/aws/xray/remote_sampler.go

```diff
@@ -83,6 +93,54 @@ func NewRemoteSampler(ctx context.Context, serviceName string, cloudPlatform str
 	return remoteSampler, nil
 }
 
+// NewRemoteSamplerWithResource returns a sampler which decides to sample a given request or not
+// based on the sampling rules set by users on AWS X-Ray console. Sampler also periodically polls
+// sampling rules and sampling targets.
+// NOTE: ctx passed in NewRemoteSamplerWithResource API is being used in background go routine. Cancellation to this context can kill the background go routine.
+// This takes in an OTel resource in order to configure service name and cloud platform for service rule matching. Their default values are empty strings if they are not in the resource.
+// Guide on AWS X-Ray remote sampling implementation (https://aws-otel.github.io/docs/getting-started/remote-sampling#otel-remote-sampling-implementation-caveats).
+func NewRemoteSamplerWithResource(ctx context.Context, otelResource *resource.Resource, opts ...Option) (sdktrace.Sampler, error) {
```

Comment:
Can we mark the `NewRemoteSampler` method as deprecated in the favor of this one?

Reply:
Marked as deprecated